ui/app/peers/peersTable.tsx

```diff
@@ -31,26 +32,38 @@ function PeerRow({ peer }: { peer: Peer }) {
 
 function PeersTable({ title, peers }: { title: string; peers: Peer[] }) {
   const [searchQuery, setSearchQuery] = useState<string>('');
+  const [filteredType, setFilteredType] = useState<DBType | undefined>(undefined);
   const rows = useMemo(
     () =>
       peers.filter((peer) => {
         return peer.name.toLowerCase().includes(searchQuery.toLowerCase());
+      }).filter((peer) => {
+        return (filteredType == undefined) || (peer.type == filteredType);
       }),
-    [peers, searchQuery]
+    [peers, searchQuery, filteredType]
   );
+  const allTypesOption: { value: DBType | undefined; label: string; } = { value: undefined, label: 'All' };
+  const availableTypes: { value: DBType | undefined; label: string; }[] = [...new Set(peers.flatMap((peer) => {
```

Comment:
Seems like this isn't giving us a unique list:
<img width="231" alt="Screenshot 2023-11-30 at 2 39 09 PM" src="https://github.com/PeerDB-io/peerdb/assets/65964360/6e664bc7-1fa8-46de-b195-b69586d39029">

Maybe something like:
```TypeScript
Array.from(
    new Map( // Map filters out duplicates
      peers.flatMap(peer => [
        [peer.type, { value: peer.type, label: DBTypeToGoodText(peer.type) }]
      ])
    ).values()
  );
```
will be better?